### ISSUE TITLE
When one worker node(out of 5) is Powered OFF and then upgrade is initiated, upgrade fails. [FATA[0104] [workerPlane] Failed to upgrade Worker Plane: [host NW2 not ready] ]

### ISSUE THREAD

**Issue by soumyalj** [rancher/rke#1909]
RKE version: v1.1.0-rc5

Docker version: (docker version,docker info preferred) 19.03

Operating system and kernel: (cat /etc/os-release, uname -r preferred) Ubuntu

**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**EC2

Steps to Reproduce:

1. Create a RKE Cluster with 3 control planes, 1 etcd node and 5 worker nodes(NW1 to NW5)
2. Power OFF one worker node NW2
3. Upgrade to a new K8s version v1.17.2-rancher1-1 using the below upgrade strategy

```
kubernetes_version: v1.17.2-rancher1-1

upgrade_strategy:
  max_unavailable: 2
  drain: true
  node_drain_input:
    force: false
    ignore_daemonsets: true
    delete_local_data: false
    grace_period: -1
    timeout: 120

```

Upgrade exits though with the below error.  There are 4 nodes which are still Ready. 
```
INFO[0104] First checking and processing worker components for upgrades on nodes with etcd/controlplane roles one at a time 
INFO[0104] [workerplane] Processing host abcd (control plane host )
ERRO[0104] Failed to upgrade hosts: NW2(worker node) with error [host NW2 not ready] 
FATA[0104] [workerPlane] Failed to upgrade Worker Plane: [host NW2 not ready] 

```
Also, in the above case, it was observed that etcd plane was upgraded, but control plane nodes did not get upgraded. Only the state was saved. While processing the worker plane components on the control/etcd planes, the upgrade exits. RKE logs below
```
WARN[0000] This is not an officially supported version (v1.1.0-rc5) of RKE. Please download the latest official release at https://github.com/rancher/rke/releases/latest 
INFO[0000] Running RKE version: v1.1.0-rc5              
INFO[0000] Initiating Kubernetes cluster                
INFO[0000] [certificates] Generating admin certificates and kubeconfig 
INFO[0000] Successfully Deployed state file at [./clusterzeroworker.rkestate] 
INFO[0000] Building Kubernetes cluster                  
INFO[0000] [dialer] Setup tunnel for host [efgh] 
INFO[0000] [dialer] Setup tunnel for host [ijkl] 
INFO[0000] [dialer] Setup tunnel for host [mnop] 
INFO[0000] [dialer] Setup tunnel for host [sdefs] 
INFO[0000] [dialer] Setup tunnel for host [qrst] 
INFO[0000] [dialer] Setup tunnel for host [jjjk] 
INFO[0000] [dialer] Setup tunnel for host [ffgh] 
INFO[0000] [dialer] Setup tunnel for host [abcd] 
INFO[0000] [dialer] Setup tunnel for host [uvwx] 
WARN[0041] Failed to set up SSH tunneling for host [sdefs]: Can't retrieve Docker Info: error during connect: Get http://%2Fvar%2Frun%2Fdocker.sock/v1.24/info: Unable to access node with address [sdefs:22] using SSH. Please check if the node is up and is accepting SSH connections or check network policies and firewall rules. Error: dial tcp sdefs:22: connect: operation timed out 
WARN[0041] Removing host [sdefs] from node lists 
INFO[0041] 2 worker nodes can be unavailable at a time  
INFO[0041] [network] No hosts added existing cluster, skipping port check 
INFO[0041] [certificates] Deploying kubernetes certificates to Cluster nodes 
INFO[0041] Checking if container [cert-deployer] is running on host [qrst], try #1    
INFO[0041] Checking if container [cert-deployer] is running on host [mnop], try #1    
INFO[0041] Checking if container [cert-deployer] is running on host [abcd], try #1    
INFO[0041] Checking if container [cert-deployer] is running on host [efgh], try #1    
INFO[0041] Checking if container [cert-deployer] is running on host [ffgh], try #1    
INFO[0041] Checking if container [cert-deployer] is running on host [jjjk], try #1    
INFO[0041] Checking if container [cert-deployer] is running on host [uvwx], try #1    
INFO[0041] Checking if container [cert-deployer] is running on host [ijkl], try #1    
INFO[0041] Image [rancher/rke-tools:v0.1.52] exists on host [abcd] 
INFO[0041] Image [rancher/rke-tools:v0.1.52] exists on host [ijkl] 
INFO[0041] Image [rancher/rke-tools:v0.1.52] exists on host [ffgh] 
INFO[0041] Image [rancher/rke-tools:v0.1.52] exists on host [jjjk] 
INFO[0041] Image [rancher/rke-tools:v0.1.52] exists on host [qrst] 
INFO[0041] Image [rancher/rke-tools:v0.1.52] exists on host [mnop] 
INFO[0041] Image [rancher/rke-tools:v0.1.52] exists on host [uvwx] 
INFO[0041] Image [rancher/rke-tools:v0.1.52] exists on host [efgh] 
INFO[0041] Starting container [cert-deployer] on host [abcd], try #1    
INFO[0041] Starting container [cert-deployer] on host [ijkl], try #1    
INFO[0041] Starting container [cert-deployer] on host [mnop], try #1    
INFO[0041] Starting container [cert-deployer] on host [jjjk], try #1    
INFO[0041] Starting container [cert-deployer] on host [qrst], try #1    
INFO[0041] Starting container [cert-deployer] on host [ffgh], try #1    
INFO[0041] Starting container [cert-deployer] on host [efgh], try #1    
INFO[0041] Starting container [cert-deployer] on host [uvwx], try #1    
INFO[0042] Checking if container [cert-deployer] is running on host [ijkl], try #1    
INFO[0042] Checking if container [cert-deployer] is running on host [jjjk], try #1    
INFO[0042] Checking if container [cert-deployer] is running on host [abcd], try #1    
INFO[0042] Checking if container [cert-deployer] is running on host [mnop], try #1    
INFO[0042] Checking if container [cert-deployer] is running on host [qrst], try #1    
INFO[0042] Checking if container [cert-deployer] is running on host [ffgh], try #1    
INFO[0042] Checking if container [cert-deployer] is running on host [uvwx], try #1    
INFO[0042] Checking if container [cert-deployer] is running on host [efgh], try #1    
INFO[0047] Checking if container [cert-deployer] is running on host [abcd], try #1    
INFO[0047] Checking if container [cert-deployer] is running on host [ijkl], try #1    
INFO[0047] Checking if container [cert-deployer] is running on host [jjjk], try #1    
INFO[0047] Checking if container [cert-deployer] is running on host [mnop], try #1    
INFO[0047] Checking if container [cert-deployer] is running on host [qrst], try #1    
INFO[0047] Checking if container [cert-deployer] is running on host [ffgh], try #1    
INFO[0047] Checking if container [cert-deployer] is running on host [uvwx], try #1    
INFO[0047] Checking if container [cert-deployer] is running on host [efgh], try #1    
INFO[0047] Removing container [cert-deployer] on host [abcd], try #1    
INFO[0047] Removing container [cert-deployer] on host [jjjk], try #1    
INFO[0047] Removing container [cert-deployer] on host [ijkl], try #1    
INFO[0047] Removing container [cert-deployer] on host [ffgh], try #1    
INFO[0047] Removing container [cert-deployer] on host [qrst], try #1    
INFO[0047] Removing container [cert-deployer] on host [mnop], try #1    
INFO[0047] Removing container [cert-deployer] on host [uvwx], try #1    
INFO[0047] Removing container [cert-deployer] on host [efgh], try #1    
INFO[0048] [reconcile] Rebuilding and updating local kube config 
INFO[0048] Successfully Deployed local admin kubeconfig at [./kube_config_clusterzeroworker.yml] 
INFO[0048] [reconcile] host [ffgh] is active master on the cluster 
INFO[0048] [certificates] Successfully deployed kubernetes certificates to Cluster nodes 
INFO[0048] [reconcile] Reconciling cluster state        
INFO[0048] [reconcile] Check etcd hosts to be deleted   
INFO[0048] [reconcile] Check etcd hosts to be added     
INFO[0048] [reconcile] Rebuilding and updating local kube config 
INFO[0048] Successfully Deployed local admin kubeconfig at [./kube_config_clusterzeroworker.yml] 
INFO[0048] [reconcile] host [ffgh] is active master on the cluster 
INFO[0048] [reconcile] Reconciled cluster state successfully 
INFO[0048] Pre-pulling kubernetes images                
INFO[0048] Pulling image [rancher/hyperkube:v1.17.2-rancher1] on host [ffgh], try #1    
INFO[0048] Pulling image [rancher/hyperkube:v1.17.2-rancher1] on host [ijkl], try #1    
INFO[0048] Pulling image [rancher/hyperkube:v1.17.2-rancher1] on host [jjjk], try #1    
INFO[0048] Pulling image [rancher/hyperkube:v1.17.2-rancher1] on host [abcd], try #1    
INFO[0048] Pulling image [rancher/hyperkube:v1.17.2-rancher1] on host [efgh], try #1    
INFO[0048] Pulling image [rancher/hyperkube:v1.17.2-rancher1] on host [mnop], try #1    
INFO[0048] Pulling image [rancher/hyperkube:v1.17.2-rancher1] on host [uvwx], try #1    
INFO[0048] Pulling image [rancher/hyperkube:v1.17.2-rancher1] on host [qrst], try #1    
INFO[0073] Image [rancher/hyperkube:v1.17.2-rancher1] exists on host [qrst] 
INFO[0073] Image [rancher/hyperkube:v1.17.2-rancher1] exists on host [abcd] 
INFO[0073] Image [rancher/hyperkube:v1.17.2-rancher1] exists on host [uvwx] 
INFO[0073] Image [rancher/hyperkube:v1.17.2-rancher1] exists on host [efgh] 
INFO[0073] Image [rancher/hyperkube:v1.17.2-rancher1] exists on host [ffgh] 
INFO[0073] Image [rancher/hyperkube:v1.17.2-rancher1] exists on host [jjjk] 
INFO[0074] Image [rancher/hyperkube:v1.17.2-rancher1] exists on host [mnop] 
INFO[0075] Image [rancher/hyperkube:v1.17.2-rancher1] exists on host [ijkl] 
INFO[0075] Kubernetes images pulled successfully        
INFO[0075] [etcd] Building up etcd plane..              
INFO[0075] Image [rancher/rke-tools:v0.1.52] exists on host [abcd] 
INFO[0078] Starting container [etcd-fix-perm] on host [abcd], try #1    
INFO[0079] Successfully started [etcd-fix-perm] container on host [abcd] 
INFO[0079] Waiting for [etcd-fix-perm] container to exit on host [abcd] 
INFO[0079] Waiting for [etcd-fix-perm] container to exit on host [abcd] 
INFO[0079] Removing container [etcd-fix-perm] on host [abcd], try #1    
INFO[0079] [remove/etcd-fix-perm] Successfully removed container on host [abcd] 
INFO[0079] Checking if container [etcd] is running on host [abcd], try #1    
INFO[0080] Pulling image [rancher/coreos-etcd:v3.4.3-rancher1] on host [abcd], try #1    
INFO[0083] Image [rancher/coreos-etcd:v3.4.3-rancher1] exists on host [abcd] 
INFO[0083] Checking if container [old-etcd] is running on host [abcd], try #1    
INFO[0083] Stopping container [etcd] on host [abcd] with stopTimeoutDuration [5s], try #1    
INFO[0084] Waiting for [etcd] container to exit on host [abcd] 
INFO[0084] Renaming container [etcd] to [old-etcd] on host [abcd], try #1    
INFO[0084] Starting container [etcd] on host [abcd], try #1    
INFO[0085] [etcd] Successfully updated [etcd] container on host [abcd] 
INFO[0085] Removing container [old-etcd] on host [abcd], try #1    
INFO[0086] [etcd] Running rolling snapshot container [etcd-snapshot-once] on host [abcd] 
INFO[0086] Removing container [etcd-rolling-snapshots] on host [abcd], try #1    
INFO[0087] [remove/etcd-rolling-snapshots] Successfully removed container on host [abcd] 
INFO[0087] Image [rancher/rke-tools:v0.1.52] exists on host [abcd] 
INFO[0088] Starting container [etcd-rolling-snapshots] on host [abcd], try #1    
INFO[0088] [etcd] Successfully started [etcd-rolling-snapshots] container on host [abcd] 
INFO[0093] Image [rancher/rke-tools:v0.1.52] exists on host [abcd] 
INFO[0094] Starting container [rke-bundle-cert] on host [abcd], try #1    
INFO[0094] [certificates] Successfully started [rke-bundle-cert] container on host [abcd] 
INFO[0094] Waiting for [rke-bundle-cert] container to exit on host [abcd] 
INFO[0095] [certificates] successfully saved certificate bundle [/opt/rke/etcd-snapshots//pki.bundle.tar.gz] on host [abcd] 
INFO[0095] Removing container [rke-bundle-cert] on host [abcd], try #1    
INFO[0095] Image [rancher/rke-tools:v0.1.52] exists on host [abcd] 
INFO[0095] Starting container [rke-log-linker] on host [abcd], try #1    
INFO[0096] [etcd] Successfully started [rke-log-linker] container on host [abcd] 
INFO[0096] Removing container [rke-log-linker] on host [abcd], try rancher/rke#1 
INFO[0096] [remove/rke-log-linker] Successfully removed container on host [abcd] 
INFO[0096] [etcd] Successfully started etcd plane.. Checking etcd cluster health 
INFO[0098] [controlplane] Processing control plane components for upgrade one at a time 
INFO[0098] Processing controlplane host ffgh    
INFO[0098] [authz] Creating rke-job-deployer ServiceAccount 
INFO[0101] [authz] rke-job-deployer ServiceAccount created successfully 
INFO[0101] [authz] Creating system:node ClusterRoleBinding 
INFO[0101] [authz] system:node ClusterRoleBinding created successfully 
INFO[0101] [authz] Creating kube-apiserver proxy ClusterRole and ClusterRoleBinding 
INFO[0102] [authz] kube-apiserver proxy ClusterRole and ClusterRoleBinding created successfully 
INFO[0102] Successfully Deployed state file at [./clusterzeroworker.rkestate] 
INFO[0102] [state] Saving full cluster state to Kubernetes 
INFO[0103] [state] Successfully Saved full cluster state to Kubernetes ConfigMap: cluster-state 
INFO[0103] [worker] Upgrading Worker Plane..            
INFO[0103] 2 worker nodes can be unavailable at a time  
INFO[0104] First checking and processing worker components for upgrades on nodes with etcd/controlplane roles one at a time 
INFO[0104] [workerplane] Processing host jjjk  
ERRO[0104] Failed to upgrade hosts: sdefs with error [host sdefs not ready] 
FATA[0104] [workerPlane] Failed to upgrade Worker Plane: [host sdefs not ready] 



```

**Comment by mrajashree** [comment]
Available with v1.1.0-rc7

**Comment by soumyalj** [comment]
Tested with RKE version v1.1.0-rc7

1. Create a RKE Cluster with 3 control planes, 1 etcd node and 5 worker nodes(NW1 to NW5) with k8s version v1.16.7-rancher1-1
2. Power OFF one worker node NW2
3. Upgrade to a new K8s version v1.17.3-rancher1-1 using the below upgrade strategy

```
kubernetes_version: v1.17.3-rancher1-1

upgrade_strategy:
  max_unavailable: 2
  drain: true
  node_drain_input:
     force: false
     ignore_daemonsets: true
     delete_local_data: false
     grace_period: -1
     timeout: 120
```

Upgrade continues further without any error. Worker nodes other than the powered OFF node upgrade successfully. Logs provide a message that the powered OFF node was skipped as below:

```
INFO[0332] [ingress] Setting up nginx ingress controller 
INFO[0332] [addons] Saving ConfigMap for addon rke-ingress-controller to Kubernetes 
INFO[0332] [addons] Successfully saved ConfigMap for addon rke-ingress-controller to Kubernetes 
INFO[0332] [addons] Executing deploy job rke-ingress-controller 
INFO[0333] [ingress] ingress controller nginx deployed successfully 
INFO[0333] [addons] Setting up user addons              
INFO[0333] [addons] no user addons defined              
FATA[0333] Provisioning incomplete, host(s) [abcd] skipped because they could not be contacted 


```
